src/okta/fetchBatchOfResources.ts

```diff
@@ -0,0 +1,156 @@
+import {
+  IntegrationCacheEntry,
+  IntegrationInstanceAuthorizationError,
+  IntegrationLogger,
+} from '@jupiterone/jupiter-managed-integration-sdk';
+import {
+  OktaClient,
+  OktaCollection,
+  OktaQueryParams,
+  OktaResource,
+} from '../okta/types';
+import {
+  OktaExecutionContext,
+  OktaIntegrationStepIterationState,
+} from '../types';
+import extractCursorFromNextUri from '../util/extractCursorFromNextUri';
+import logIfForbiddenOrNotFound from '../util/logIfForbidden';
+import retryApiCall from '../util/retryApiCall';
+import { OktaCacheState } from './types';
+
+/**
+ * An iterating execution handler that loads Okta resources and their associated
+ * data in `BATCH_PAGES` batches of `PAGE_LIMIT` resources, storing the raw
+ * response data in the `IntegrationCache` for later processing in another step.
+ *
+ * This is necessary because Okta throttles API requests, leading to a need to
+ * spread requests over a period of time that exceeds the execution time limits
+ * of the execution environment.
+ */
+export default async function fetchBatchOfResources<
```

Comment:
Same here. I'm sure this will inform the new implementation of `iterate*` functions, but the new code will be far simpler.

Reply:
Removed